### DISCUSSION TITLE
DecoratorNode without React

### DISCUSSION THREAD

**Discussion by mkdynamic** [facebook/lexical#4715] [Q&A]
All the examples have decorate() return a JSX.Element. However, I cannot find where that is actually mounted into the DOM.

My understanding is the createDOM() HTMLElement will be inserted into the DOM, but I do not understand how the decorate() function interacts.

Specifically, I want to create a custom DecoratorNode and attach functionality you the mounted DOM node but without React, just Vanilla JS. Returning an HTMLElement from decorate() has no effect.

Any guidance appreciated.

**Comment by placeba** [comment]
This article describes how DecoratorNode works. 
https://github.com/umaranis/svelte-lexical/blob/master/docs/decorator-node.md

**Reply by talaat197** [reply]
I had the same problem and the solution is pretty simple use the DecoratorNode and add the following line of codes

```
  decorate() {
    return null;
  }
```

**Reply by etrepum** [reply]
That static doesn’t do anything, returning null is sufficient.
